src/query/server/server.go

```diff
@@ -727,25 +736,40 @@ func remoteClient(
 }
 
 func startGrpcServer(
-	logger *zap.Logger,
 	storage m3.Storage,
+	queryContextOptions models.QueryContextOptions,
 	poolWrapper *pools.PoolWrapper,
 	cfg *config.RPCConfiguration,
+	logger *zap.Logger,
 ) (*grpc.Server, error) {
 	logger.Info("creating gRPC server")
-	server := tsdbRemote.CreateNewGrpcServer(storage, poolWrapper)
-	waitForStart := make(chan struct{})
-	var startErr error
+	server := tsdbRemote.CreateNewGrpcServer(storage,
+		queryContextOptions, poolWrapper)
+
+	var (
+		waitForStart = make(chan struct{}, 1)
+		startErr     atomic.Value
+		wg           sync.WaitGroup
+	)
+	startErr.Store(nil)
+	wg.Add(1)
```

Comment:
I dont think you actually use this

Reply:
Ta, I'm going to rip all this stuff out thankfully - good catch though.